### PR TITLE
server: move RegionInfo to core directory.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: server/core/region.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+package core
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+)
+
+// RegionInfo record detail region info
```
Comment (huachaohuang) [inline server/core/region.go]: s/record detail/records detailed/

File: server/core/region.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	return &RegionInfo{
+		Region:       proto.Clone(r.Region).(*metapb.Region),
+		Leader:       proto.Clone(r.Leader).(*metapb.Peer),
+		DownPeers:    downPeers,
+		PendingPeers: pendingPeers,
+		WrittenBytes: r.WrittenBytes,
+	}
+}
+
+// GetPeer return the peer with specified peer id
```
Comment (huachaohuang) [inline server/core/region.go]: s/return/returns

File: server/core/region.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+func (r *RegionInfo) GetPeer(peerID uint64) *metapb.Peer {
+	for _, peer := range r.GetPeers() {
+		if peer.GetId() == peerID {
+			return peer
+		}
+	}
+	return nil
+}
+
+// GetDownPeer return the down peers with specified peer id
```
Comment (huachaohuang) [inline server/core/region.go]: ditto

**Comment by siddontang** [comment]
CI failed

**Pull request merged**